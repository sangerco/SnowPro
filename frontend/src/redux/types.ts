export interface UserData {
    username: string;
    firstName: string;
    lastName: string;
    email: string;
    avatar: string;
    bio: string;
    videos: string[];
    photos: string[];
    favMountains: string[];
  }


export const FETCH_USER_DATA_REQUEST = 'FETCH_USER_DATA_REQUEST';
export const FETCH_USER_DATA_SUCCESS = 'FETCH_USER_DATA_SUCCESS';
export const FETCH_USER_DATA_FAILURE = 'FETCH_USER_DATA_FAILURE';


export interface FetchUserDataRequestAction {
    type: typeof FETCH_USER_DATA_REQUEST
}

export interface FetchUserDataSuccessAction {
    type: typeof FETCH_USER_DATA_SUCCESS;
    payload: UserData;
}

export interface FetchUserDataFailureAction {
    type: typeof FETCH_USER_DATA_FAILURE;
    payload: string;
}

export type UserActionTypes = 
    FetchUserDataRequestAction
    | FetchUserDataSuccessAction
    | FetchUserDataFailureAction;