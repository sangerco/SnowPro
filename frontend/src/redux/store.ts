import { Action, compose } from "redux";
import { ThunkAction, configureStore } from "@reduxjs/toolkit";
import { logger } from "redux-logger";
import rootReducer from "./reducers/rootReducer";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(logger, composeEnhancers),
});
