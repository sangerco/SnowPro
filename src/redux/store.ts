import { configureStore, combineReducers } from "@reduxjs/toolkit";
import userReducer from "./slices/userSlice";
import authReducer from "./slices/authSlice";
import messageReducer from "./slices/messageSlice";
import messageReplyReducer from "./slices/messageReplySlice";
import reviewReducer from "./slices/reviewSlice";
import reviewReplyReducer from "./slices/reviewReplySlice";
import tagReducer from "./slices/tagSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";

const persistConfig = {
  key: "root",
  storage,
  blacklist: ["messages", "messageReplies", "reviews", "reviewReplies"],
};

const rootReducer = combineReducers({
  users: userReducer,
  auth: authReducer,
  messages: messageReducer,
  messageReplies: messageReplyReducer,
  reviews: reviewReducer,
  reviewReplies: reviewReplyReducer,
  tags: tagReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const persistor = persistStore(store);
