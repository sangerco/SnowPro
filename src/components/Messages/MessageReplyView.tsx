import React, { useState } from "react";
import { Button, Card, Icon, Modal } from "semantic-ui-react";
import { deleteMessageReply } from "../../redux/slices/messageReplySlice";
import { useDispatch } from "react-redux";
import { AppDispatch } from "../../redux/store";
import { MessageReplyData } from "../../redux/slices/messageReplySlice";
import { Link } from "react-router-dom";

interface MessageReplyViewProps {
  messageReply: MessageReplyData;
}

const MessageReplyView: React.FC<MessageReplyViewProps> = ({
  messageReply,
}) => {
  const dispatch = useDispatch<AppDispatch>();
  const [showDeleteModal, setShowDeleteModal] = useState(false);

  const handleShowDeleteModal = () => {
    setShowDeleteModal(true);
  };

  const handleDeleteMessageReply = () => {
    if (messageReply.id) {
      dispatch(deleteMessageReply(messageReply.id));
    }
  };

  const formatDate = (date: Date) => {
    const newDate = new Date(date);
    const options: Intl.DateTimeFormatOptions = {
      year: "numeric",
      month: "long",
      day: "numeric",
    };
    return newDate.toLocaleDateString("en-US", options);
  };

  return (
    <div>
      <Card
        data-testid="message-reply"
        as={Link}
        to={`/messages/replies/${messageReply.id}`}
        style={{ width: "500px" }}>
        <Card.Header>{messageReply.subject}</Card.Header>
        <Card.Meta textAlign="right">
          {formatDate(messageReply.createdAt)}
        </Card.Meta>
        <Card.Meta>
          {messageReply.senderFirstName} {messageReply.senderLastName}
        </Card.Meta>
        <Card.Content data-testid="message-reply-body">
          {messageReply.body}
        </Card.Content>
        <Card.Content>
          <Button
            basic
            color="red"
            onClick={handleShowDeleteModal}
            data-testid="delete-button">
            <Icon name="trash" />
          </Button>
        </Card.Content>
      </Card>
      <Modal
        open={showDeleteModal}
        onClose={() => setShowDeleteModal(false)}
        data-testid="delete-modal">
        <Modal.Content>
          Are You Sure You Want To Delete This Message Reply?
        </Modal.Content>
        <Modal.Actions>
          <Button negative onClick={handleDeleteMessageReply}>
            Yes
          </Button>
          <Button onClick={() => setShowDeleteModal(false)}>Cancel</Button>
        </Modal.Actions>
      </Modal>
    </div>
  );
};

export default MessageReplyView;
