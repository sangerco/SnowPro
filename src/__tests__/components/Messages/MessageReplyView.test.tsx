/* eslint-disable testing-library/no-node-access */
import React from "react";
import MessageReplyView from "../../../components/Messages/MessageReplyView";
import { render, screen, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";
import { BrowserRouter } from "react-router-dom";

const messageReplyData = {
  id: "testId",
  messageId: "messageTestId",
  senderId: "11",
  recipientId: "22",
  subject: "test message reply",
  createdAt: new Date(),
  senderUsername: "testuser",
  senderFirstName: "Test",
  senderLastName: "User",
  body: " message reply test",
  recipientUsername: "testuser2",
  recipientFirstName: "Test",
  recipientLastName: "Name2",
};

describe("test Message Reply View", () => {
  it("should render without crashing", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <MessageReplyView messageReply={messageReplyData} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <MessageReplyView messageReply={messageReplyData} />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
