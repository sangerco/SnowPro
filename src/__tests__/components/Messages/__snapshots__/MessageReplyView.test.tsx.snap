// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Message Reply View should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="ui card"
      href="/messages/replies/testId"
      style="width: 500px;"
    >
      <div
        class="header"
      >
        test message reply
      </div>
      <div
        class="right aligned meta"
      >
        October 1, 2023
      </div>
      <div
        class="meta"
      >
        Test User
      </div>
      <div
        class="content"
      >
         message reply test
      </div>
      <div
        class="content"
      >
        <button
          class="ui red basic button"
        >
          <i
            aria-hidden="true"
            class="trash icon"
          />
        </button>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
